### ISSUE TITLE
Rendering in Vitest results in `undefined (reading 'call')`

### ISSUE THREAD

**Issue by pauleveritt** [jeasx/jsx-async-runtime#3]
*Let me know if you need a repo.* I'm trying out 0.4. This code works fine in `foo.tsx` when run from `tsx`:

```tsx
import { jsxToString } from "jsx-async-runtime";

function Content() {
  return <div>yes</div>;
}

const content = <Content />;
const result = await jsxToString(content);

console.log(result);
```

When I put this in a Vitest test:

```tsx
import { test } from "vitest";
import { jsxToString } from "jsx-async-runtime";

test("renders foo", async () => {
  function Content() {
    return <div>yes</div>;
  }

  const content = <Content />;
  const result = await jsxToString(content);
});
```

I get:

```
TypeError: Cannot read properties of undefined (reading 'call')
    at Module.o (file:///Users/pauleveritt/projects/pauleveritt/www.pauleveritt.org/node_modules/jsx-async-runtime/dist/esm/index.js:113:29)
    at Module.o (file:///Users/pauleveritt/projects/pauleveritt/www.pauleveritt.org/node_modules/jsx-async-runtime/dist/esm/index.js:107:33)
```

It fails on [this line](https://github.com/jeasx/jsx-async-runtime/blob/main/src/render/jsx-to-string.ts#L87)...but the second time through. The first time, `t.tag` is `function Content()`. I continue and at the breakpoint the second time, there is no `t.tag`. Screenshot attached of what it is stopped on.

![bug](https://github.com/jeasx/jsx-async-runtime/assets/455142/bff55fa1-eb29-43f2-b281-6ec738c9c238)

**Comment by jablonski** [comment]
Can you provide a repo so I can check with Vitest? Seems strange...

**Comment by pauleveritt** [comment]
Here's a repo with some instructions in README: https://github.com/pauleveritt/vitest-jsx-async-runtime

**Comment by jablonski** [comment]
Repo seems to be private...;-)

**Comment by pauleveritt** [comment]
@jablonski Crap, that's the second time I've done this. I make the repo from WebStorm and forget it defaults to private. Fixed, sorry.

**Comment by jablonski** [comment]
Hi Paul,

thanks for reporting.

Vitest injects an empty object ({}) as JsxElement, which wasn't handled properly by jsx-async-runtime. This is fixed in 0.4.1.

IMHO injecting an empty object as JsxElement looks like a Vitest bug too me. At least I cannot think of any good reason for this behaviour, but it is good to have it covered in jsx-async-runtime anyway.

**Comment by pauleveritt** [comment]
I'm happy to say, I plugged it in, all my tests are passing.
